polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/PageTokenUtil.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.persistence.pagination;
+
+import static com.google.api.client.util.Preconditions.checkArgument;
+
+import jakarta.annotation.Nullable;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+final class PageTokenUtil {
+
+  private PageTokenUtil() {}
+
+  static PageToken decodePageRequest(
+      @Nullable String requestedPageToken, @Nullable Integer requestedPageSize) {
+    int pageSize;
+    String encodedDataReference = null;
+    if (requestedPageToken != null) {
+      byte[] bytes = Base64.getUrlDecoder().decode(requestedPageToken);
+      String token = new String(bytes, StandardCharsets.UTF_8);
+      int idx = token.indexOf(':');
+      checkArgument(idx > 0, "Invalid page token: %s", requestedPageToken);
+      int prevPageSize = Integer.parseInt(token.substring(0, idx));
+      encodedDataReference = token.substring(idx + 1);
+      pageSize = Optional.ofNullable(requestedPageSize).orElse(prevPageSize);
+    } else {
+      pageSize = Optional.ofNullable(requestedPageSize).orElse(-1);
+    }
+
+    return new PageToken(encodedDataReference, pageSize);
+  }
+
+  static @Nullable String encodePageToken(
```

Comment:
Why is this not just a member of `PageToken`?

Reply:
To co-locate encoding and decoding code in one java file.

Reply:
`decodePageRequest` is called from `PageToken` when the token is received, but `encodePageToken` is called from`Page` when it is returned to the client. This class is the place to share code between `PageToken` and `Page`.